tests/types/npm-test.json

```diff
@@ -254,6 +254,316 @@
       "expected_output": "pkg:npm/core@2.0.1#googleapis/api/annotations",
       "expected_failure": false,
       "expected_failure_reason": null
+    },
+    {
+      "description": "Negative Test: An npm purl with an un-encoded @ in scope should fail parsing",
+      "test_group": "advanced",
+      "test_type": "parse",
+      "input": "pkg:npm/@angular/animation@12.3.1",
+      "expected_output": null,
+      "expected_failure": true,
+      "expected_failure_reason": "Should fail to parse due to unencoded '@' in the namespace"
+    },
+    {
+      "description": "Negative Test: An npm purl with an uppercase name should fail parsing",
+      "test_group": "advanced",
+      "test_type": "parse",
+      "input": "pkg:npm/React@17.0.1",
+      "expected_output": null,
+      "expected_failure": true,
+      "expected_failure_reason": "Should fail to parse due to uppercase characters in the name"
+    },
+    {
+      "description": "Normalization Test: An npm purl with an uppercase name should be lowercased on roundtrip",
+      "test_group": "advanced",
+      "test_type": "roundtrip",
+      "input": "pkg:npm/LoDash@4.17.21",
+      "expected_output": "pkg:npm/lodash@4.17.21",
+      "expected_failure": false,
+      "expected_failure_reason": null
+    },
```

Comment:
This test is wrong, and also conflicts with the previous test. The correct output is `pkg:npm/LoDash`.